src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -161,28 +181,110 @@ protected ExecutionStrategy(DataFetcherExceptionHandler dataFetcherExceptionHand
      *
      * @return a promise to an {@link ExecutionResult}
      *
-     * @throws NonNullableFieldWasNullException in the future if a non null field resolves to a null value
+     * @throws NonNullableFieldWasNullException in the future if a non-null field resolves to a null value
      */
     public abstract CompletableFuture<ExecutionResult> execute(ExecutionContext executionContext, ExecutionStrategyParameters parameters) throws NonNullableFieldWasNullException;
 
+    /**
+     * This is the re-entry point for an execution strategy when an object type needs to be resolved.
+     *
+     * @param executionContext contains the top level execution parameters
+     * @param parameters       contains the parameters holding the fields to be executed and source object
+     *
+     * @return a promise to an {@link Map<String,Object>}
+     *
+     * @throws NonNullableFieldWasNullException in the future if a non-null field resolves to a null value
+     */
```

Comment:
This is our call back for ES -> ES used when running `completeObject` calls - it returns an object and NOT a `ExecutionResult` like `es.execute` used to do